src/librustc_mir/transform/inline.rs

```diff
@@ -125,6 +125,13 @@ impl Inliner<'tcx> {
                     continue;
                 };
 
+                // Copy unevaluated constants
+                let mut callee_uneval_consts = callee_body.uneval_consts.clone();
+                callee_uneval_consts.retain(|&constant| {
+                    matches!(constant.literal.val, ConstKind::Unevaluated(_, _, _))
+                });
+                caller_body.uneval_consts.extend(callee_uneval_consts);
```

Comment:
```suggestion
                caller_body.uneval_consts.extend(callee_body.uneval_consts.iter().copied().filter(|&constant| {
                    matches!(constant.literal.val, ConstKind::Unevaluated(_, _, _))
                });
```

Reply:
Thanks, I've done a rebase on my branch and applied this.